utils/src/lib.rs

```diff
@@ -531,123 +531,101 @@ pub fn encode_args(func: &Function, args: &[impl AsRef<str>]) -> Result<Vec<u8>>
-    let res = reqwest::get(url).await?;
-    let res = res.text().await?;
+    // using samczsun signature database over 4byte
+    // see https://github.com/foundry-rs/foundry/issues/1672
+    let base_url = "https://sig.eth.samczsun.com/api/v1/signatures";
+    let url = match selector_type {
+        SelectorType::Function => format!("{base_url}?function={selector}"),
+        SelectorType::Event => format!("{base_url}?event={selector}"),
+    };
+
+    let res = reqwest::get(url).await?.text().await?;
     let api_response = match serde_json::from_str::<ApiResponse>(&res) {
         Ok(inner) => inner,
         Err(err) => {
             eyre::bail!("Could not decode response:\n {res}.\nError: {err}")
         }
     };
 
-    if api_response.results.is_empty() {
-        eyre::bail!("no signature found for provided function selector")
+    if !api_response.ok {
+        eyre::bail!("Failed to decode:\n {res}")
     }
 
-    Ok(api_response
-        .results
+    let decoded = match selector_type {
+        SelectorType::Function => api_response.result.function,
+        SelectorType::Event => api_response.result.event,
+    };
+
+    Ok(decoded
+        .get(selector)
+        .ok_or(eyre::eyre!("No signature found"))?
         .into_iter()
-        .map(|d| (d.text_signature, d.id))
-        .collect::<Vec<(String, i32)>>())
+        .map(|d| d.name.clone())
+        .collect::<Vec<String>>())
 }
 
-pub async fn fourbyte_possible_sigs(calldata: &str, id: Option<String>) -> Result<Vec<String>> {
-    let mut sigs = fourbyte(calldata).await?;
-
-    match id {
-        Some(id) => {
-            let sig = match &id[..] {
-                "earliest" => {
-                    sigs.sort_by(|a, b| a.1.cmp(&b.1));
-                    sigs.get(0)
-                }
-                "latest" => {
-                    sigs.sort_by(|a, b| b.1.cmp(&a.1));
-                    sigs.get(0)
-                }
-                _ => {
-                    let id: i32 = id.parse().expect("Must be integer");
-                    sigs = sigs
-                        .iter()
-                        .filter(|sig| sig.1 == id)
-                        .cloned()
-                        .collect::<Vec<(String, i32)>>();
-                    sigs.get(0)
-                }
-            };
-            match sig {
-                Some(sig) => Ok(vec![sig.clone().0]),
-                None => Ok(vec![]),
-            }
-        }
-        None => {
-            // filter for signatures that can be decoded
-            Ok(sigs
-                .iter()
-                .map(|sig| sig.clone().0)
-                .filter(|sig| {
-                    let res = abi_decode(sig, calldata, true);
-                    res.is_ok()
-                })
-                .collect::<Vec<String>>())
-        }
+/// Fetches a function signature given the selector using sig.eth.samczsun.com
+pub async fn fourbyte(selector: &str) -> Result<Vec<String>> {
```

Comment:
we should either rename this function or duplicate it for sig.eth.samczsun

Reply:
renamed to a more generic `decode*` format; kept the CLI command naming the same though for user continuity